### ISSUE TITLE
vectorTileLayerStyles doesn't get called on load for large topojson

### ISSUE THREAD

**Issue by derhoads** [Leaflet/Leaflet.VectorGrid#194]
Hi,

First of all, thanks for the great library! 

Basically, I'm loading a large topojson (53,1 MB) and it loads with the default style (thick blue border, blue transparent fill) instead of the style I've defined in vectorTileLayerStyles. I put a flag in vectorTileLayerStyles and I can see that it just isn't getting called. 

I know it isn't a code issue, because I'm using the same code for a different topojson on a separate page and it loads fine. The only difference is the data. 

I know it's possibly a data issue, but I wanted to know basically if anyone has had a similar experience, and how they resolved it. I thought it's a bit strange that the topojson actually does load successfully, just doesn't get styled. 

Thanks in advance,
Daniel

**Comment by tomchadwin** [comment]
Probably not enough info to debug. Can you post it anywhere so we can recreate the error?

**Comment by derhoads** [comment]
Sure. Shoudl have added context.
Here's the json:
http://cosin3.rdi.uoc.edu/espacio-persona/data/mad/voronoi_tp.json

And here's the error in action...
http://cosin3.rdi.uoc.edu/espacio-persona/test.html

Relevant code:

Parse json
```javascript
$.getJSON(espaiPersona.dataPaths.voronoi, function(dat){
    espaiPersona.funcs.addTopoData(dat);
  });
```

Callback from above:
```javascript      
addTopoData: function(json) {   
    espaiPersona.layers.voronoi = L.vectorGrid.slicer( json, {
        rendererFactory: L.canvas.tile,
        maxNativeZoom: 25,
        maxZoom: 25,
        interactive: true,
        getFeatureId: function(f){
            return f.properties.id;
        },
        vectorTileLayerStyles:
        {
          voronoi: function(properties){
             //never gets called
              console.log(properties);
              var style;
              var color = espaiPersona.funcs.getCellColor(properties);
              style = espaiPersona.styles.voronoi;
              style.fillColor = color;
              style.color = color;
              return style;
          }
        }
    }).on('load',function(e){
        $("#average").text("N/A");
    }).bindPopup(function(layer){
        return espaiPersona.funcs.buildCellPopup(layer.properties.id);
    }).addTo(espaiPersona.map);
    espaiPersona.activeLayer = espaiPersona.layers.voronoi;
}
```

Styles
```javascript
espaiPersona.styles.voronoi: {
            fill: true,
            fillOpacity: 1,
            fillColor: "#ffffff",
            weight: 1,
            color: "#ffffff"
}
espaiPersona.funcs.getCellColor: function(properties){
    var prob = properties.prob;
    color = prob < 0.2 ? '#2c7bb6' :
        prob >= 0.2 && prob < 0.4 ? '#abd9e9':
        prob >= 0.4 && prob < 0.6 ? '#ffffc0':
        prob >= 0.6 && prob < 0.8 ? '#fdae61':
        '#d7191c'
    return color;
}
```

**Comment by derhoads** [comment]
So, embarassingly, this was just a stupid mistake on my part. I should have read the documentation on styling more carefully, but I'll give my solution just in case someone runs into the same issue.

Basically, I naively assumed that if I defined just one style in vectorTileLayerStyles, it would be automatically applied to the layer. That's not true. It depends on the name you give the style and the names of the layers in your file (or the name you apply to the layer in the vectorTileLayerName option of VectorGrid.slicer??). 

Interestingly, as I was trying everything that came to my mind, I tried changing the name of the style to the default vectorTileLayerName ('sliced') from the VectorGrid.slicer options. This didn't work for my topojson, but actually worked for a .geojson file with the same in-data layer name. I don't know if that means the .geojson name gets overriden, but topojson doesn't, but thought it was curious.

Anyway, this is resolved...
